src/main/java/com/example/springbootbookshop/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.example.springbootbookshop.security;
+
+import com.example.springbootbookshop.exception.EntityNotFoundException;
+import com.example.springbootbookshop.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username).orElseThrow(() ->
+                new EntityNotFoundException("User doesn`t exist"));
```

Comment:
You can throw UsernameNotFoundException here and remove it from signature

Reply:
Not fixed

Reply:
This method throw UsernameNotFoundException so you need to 
userRepository.findByEmail(username).orElseThrow(() ->
                new UsernameNotFoundException("User doesn`t exist"))